presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarballController.java

```diff
@@ -0,0 +1,138 @@
+package com.teradata.prestomanager.agent;
+
+import io.airlift.log.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.teradata.prestomanager.agent.AgentFileUtils.downloadFile;
+import static com.teradata.prestomanager.agent.PrestoConfigUtils.addConfigFiles;
+import static com.teradata.prestomanager.agent.PrestoConfigUtils.addConnectors;
+import static com.teradata.prestomanager.agent.PrestoConfigUtils.untarFile;
+import static java.lang.String.format;
+import static java.nio.file.Files.createDirectories;
+import static java.nio.file.Files.isDirectory;
+
+public class TarballController
+        extends PackageController
+{
+    private static final Logger LOGGER = Logger.get(TarballController.class);
+
+    private static final Path TAR_INSTALLATION_DIR = Paths.get("/usr/presto");
+    private static final Path TAR_DATA_DIR = Paths.get("/usr/data");
```

Comment:
I recommend `presto/{presto-installation}` as the tar installation directory, `{tar installation directory}/data` as the data directory, and `{tar installation directory}/config` as the configuration directory.

I would also call these `INSTALL_DIR`, and `DATA_DIR`; the `TAR_` is redundant in the `TarballController` class.